packages/cli/src/modules/insights/insights-pruning.service.ts

```diff
@@ -0,0 +1,51 @@
+import { Service } from '@n8n/di';
+import { Logger } from 'n8n-core';
+
+import { InsightsByPeriodRepository } from './database/repositories/insights-by-period.repository';
+import { InsightsConfig } from './insights.config';
+
+@Service()
+export class InsightsPruningService {
+	private pruneInsightsTimer: NodeJS.Timer | undefined;
+
+	constructor(
+		private readonly insightsByPeriodRepository: InsightsByPeriodRepository,
+		private readonly config: InsightsConfig,
+		private readonly logger: Logger, // Assuming a logger is injected,
+	) {
+		this.logger = this.logger.scoped('insights');
+	}
+
+	get isPruningEnabled() {
+		return this.config.maxAgeDays > -1;
+	}
+
+	startPruningTimer() {
+		if (!this.isPruningEnabled) {
+			return;
+		}
+
+		this.stopPruningTimer();
+		this.pruneInsightsTimer = setInterval(
+			async () => await this.pruneInsights(),
+			this.config.pruneCheckIntervalHours * 60 * 60 * 1000,
+		);
+		this.logger.debug(`Insights pruning every ${this.config.pruneCheckIntervalHours} hours`);
+	}
+
+	stopPruningTimer() {
+		if (this.pruneInsightsTimer !== undefined) {
+			clearInterval(this.pruneInsightsTimer);
+			this.pruneInsightsTimer = undefined;
+		}
+	}
+
+	async pruneInsights() {
+		try {
+			const result = await this.insightsByPeriodRepository.pruneOldData(this.config.maxAgeDays);
+			this.logger.debug('Hard-deleted insights', { count: result.affected });
```

Comment:
1. There is no soft-deletion of insights right? If so `Hard-deleted` sounds confusing.
2. A "count of insights" also sounds confusing, esp. as there are multiple insights tables. Is this total rows?